### ISSUE TITLE
how to define volumes and volumeMounts e.g. for kaniko

### ISSUE THREAD

**Issue by MarcoSchaar** [GoogleContainerTools/skaffold#4441]
Hi,

some Questions regarding documentation /examples:

based on Pod yaml definition, how to define volumes in Cluster section and volumeMounts for kaniko?

when "skipTLS: false", where to define the CA for successful verification?

**Comment by MarcoSchaar** [comment]
maybe related to https://github.com/GoogleContainerTools/skaffold/issues/4175

**Comment by MarcoSchaar** [comment]
as per #3883 and others 
found out that it should be lower case for volumeMounts like this 
```yaml
volumeMounts:
- name: kaniko-secret
   mountpath: /kaniko/.docker
...
  subpath: <...>
```

but for volumes it not clear
```yaml
cluster:

        namespace: <namespace>
        volumes:
        - name: kaniko-secret
          secret:
            secretName: kaniko-secret
        - name: <name>
          configMap:
            name: <name>
```

secret or configMap not known in "v1.Volume"

**Comment by briandealwis** [comment]
There's some work in progress in #4362 — please take a look and provide feedback!

**Comment by briandealwis** [comment]
Apologies for letting this drop.  The Skaffold `cluster.volumes` and `artifacts.kaniko.volumeMounts` reuse object types from the Kubernetes definitions, but these definitions aren't annotated in a way that our YAML serialization libraries can use (#4175).  As a result, all of the keys under `cluster.volumes` and `artifacts.kaniko.volumeMounts` must be lower-case, and some features like object inlining don't work.  Unfortunately the `configMap.name` is such an inlined field and so your example won't work.

As I discovered when investigating  https://github.com/GoogleContainerTools/skaffold/issues/4555#issuecomment-664491977 a side-effect of #4175 means that `skaffold fix` actually shows all of the fields of `cluster.volumes` and `artifacts.kaniko.volumeMounts`.

Closing this as it's ultimately a duplicate of #4175 and #4362.
